android/src/main/java/io/radar/flutter/RadarFlutterPlugin.java

```diff
@@ -1201,92 +1171,197 @@ public static class RadarFlutterReceiver extends RadarReceiver {
 
         @Override
         public void onEventsReceived(Context context, RadarEvent[] events, RadarUser user) {
-            RadarFlutterPlugin.initializeBackgroundEngine(context);
-
             try {
+                SharedPreferences sharedPrefs = context.getSharedPreferences(TAG, Context.MODE_PRIVATE);
+                long callbackHandle = sharedPrefs.getLong("events", 0L);
+
+                if (callbackHandle == 0L) {
+                    return;
+                }
+
+                RadarFlutterPlugin.initializeBackgroundEngine(context);
+                
                 JSONObject obj = new JSONObject();
                 obj.put("events", RadarEvent.toJson(events));
                 obj.put("user", user.toJson());
 
-                HashMap<String, Object> map = new Gson().fromJson(obj.toString(), HashMap.class);
-                if (sEventsSink != null) {
-                    sEventsSink.success(map);
+                HashMap<String, Object> res = new Gson().fromJson(obj.toString(), HashMap.class);
+                synchronized(lock) {
```

Comment:
@tjulien, I believe the previous developer attempted to block the event listener channel in a single line.

Reply:
Yeah. Does the new callback pattern require this?  Do you have links to Flutter docs handy that recommend how to do this?  Or links to other open source SDKs or apps that use the new background channel pattern?  That may help us figure out if this is required.

The reason I bring it up is that we don't use synchronized almost anywhere through the SDKs because it's risky (can lead to a deadlock), especially when doing something like calling another thread, like this code is doing.

synchronized -> call a listener method on the main thread -> // we don't control what happens next.  Could it call into Radar code that is now blocked on this synchronized block?  Just need to be very careful.